### DISCUSSION TITLE
How do I delete/replace a node in the cluster?

### DISCUSSION THREAD

**Discussion by Keyruu** [mysticaltech/terraform-hcloud-kube-hetzner#820] [Q&A] [answered]
I have nodes that unfortunately don't have the right stuff installed for longhorn. I don't really know why that is. And some Nodes get rate limited for registry.k8s.io. I would just like to replace the nodes that are broken. How do I do that?

**Comment by Keyruu** [comment]
I also have this problem now: ╷
│ Error: file provisioner error
│ 
│   with module.kube-hetzner.module.agents["1-0-ship-large"].null_resource.registries,
│   on .terraform/modules/kube-hetzner/modules/host/main.tf line 108, in resource "null_resource" "registries":
│  108:   provisioner "file" {
│ 
│ timeout - last error: <nil>

**Comment by mysticaltech** [comment]
@Keyruu You have to reach out to Hetzner support and normally they can change your affected node IP (to one that is not blacklisted by google cloud registry).

**Reply by Keyruu** [reply]
I want just want to recreate the node. Is there a way to do that?

**Reply by mysticaltech** [reply, marked as answer]
@Keyruu If you do not want to pass by support, there is a trick you can use. Drain the node with kubectl, delete it in hetzner. This will release the IP, then grab a static IP (normally this will be your old IP that is blacklisted), then apply terraform again, it will realize the node is missing and create a new one (if not use terraform replace). This will create the node with a new IP hopefully not blacklisted. Release the fixed IP to not have to pay for it for nothing. Good luck!
